server/datastore/mysql/hosts.go

```diff
@@ -1404,16 +1404,10 @@ func (ds *Datastore) filterHostsByOSSettingsStatus(sql string, opt fleet.HostLis
 	// or are servers. Similar logic could be applied to macOS hosts but is not included in this
 	// current implementation.
 
-	// TODO once testLabelsListHostsInLabelOSSettings enrolls hosts into the correct MDM, switch to this:
-	/*sqlFmt := ` AND (
+	sqlFmt := ` AND (
 		(h.platform = 'windows' AND mwe.host_uuid IS NOT NULL AND hmdm.enrolled = 1) -- windows
 		OR (h.platform IN ('darwin', 'ios', 'ipados') AND ne.id IS NOT NULL AND hmdm.enrolled = 1) -- apple
 		OR (h.platform = 'ubuntu' OR h.os_version LIKE 'Fedora%%') -- linux
```

Comment:
This piqued my interest into how/if the `LIKE` search might benefit from an index on `os_version`. As it stands, I think this might lead to a full table scan. This [blog](https://billauer.co.il/blog/2020/12/mysql-index-pattern-matching-performance/) had a nice summary for anyone else who might be interested.

Reply:
^ @mostlikelee it'd be great to get your thoughts as it looks like this is related to your recent [PR](https://github.com/fleetdm/fleet/pull/24200).

Reply:
@gillespi314 is this worth merging as-is? Or should we do more digging first?

Reply:
The platform LIKE is already live in 4.60 so at worst this isn't a regression.

If we wanted we could scope the Fedora platform check to exact-match on rhel on platform, which should avoid the full table scan. Scope on that would be small enough to include here.

Reply:
>  we could scope the Fedora platform check to exact-match on rhel on platform

@iansltx to clarify, do you mean something like

```sql
		OR (h.platform = 'ubuntu' OR (h.platform = 'rhel' AND h.os_version LIKE 'Fedora%%')) -- linux
```

or did you mean remove the Fedora check altogether?

Reply:
The former of the two. We need the Fedora check in some form because we have customers on Fedora whose hosts should show up here, but Fedora hosts are a subset of `rhel` hosts.

If we do that change and care a lot about performance, we'll want to do this tweak on the Linux disk encryption queries as well, not just here.

Reply:
> or did you mean remove the Fedora check altogether?

I think we'd want to remove the filter on the unindexed `os_version`. If `platform = 'rhel'` is substantially equivalent for purposes of the disk encryption feature, that would be ideal since we have an index already on platform.

Reply:
that's a great point, this was hurried in a bit without digging into performance

Reply:
@gillespi314 An EXPLAIN on this would be informative here; the OS platform LIKE should get filtered after using the index on platform alone so we shouldn't lose anything significant perf wise by keeping that in (it'll use the platform index).

RHEL vs. RHEL + Fedora would _not_ be equivalent here since:

1. LUKS encryption is currently limited to platforms for which we can throw a UI at the user for entering their passphrase, and those are limited to desktop Fedora plus Ubuntu right now. Non-Fedora RHEL is assumed to be headless/server, and we don't want to show a bunch of RHEL/CentOS/Rocky/Alma servers in this filter that aren't actionable.
2. Unlike with Windows/Apple, we can't pre-filter hosts by whether they're enrolled in MDM (Linux disk encryption doesn't require MDM and Linux doesn't have a concept of centralized MDM), so the platform filter is as good as we can get for constraining which hosts show up in this filter.

Of note (and out of scope for this discussion): there's another issue filed for making sure Linux hosts don't show up in OS settings filtering when disk encryption enforcement is turned off, and that'll hit these queries as well when that fix is implemented.

Reply:
> If we do that change and care a lot about performance, we'll want to do this tweak on the Linux disk encryption queries as well, not just here.

@iansltx, I think we should pursue it. Are you the best person to make those changes? The various disk encryption encryption and OS settings queries have been a source of performance issues in the past, and we've taken pains to optimize around indexes wherever possible.

Reply:
@gillespi314 Yeah, it's fine to file a released bug against EOps for this, and I'll likely be the one to pick it up. If @sharon-fdm feels strongly enough about getting this into 4.61.0 stuff can be shuffled around to make that happen.

For context, the other count/filtering related bugs around this (including the Linux one I mentioned) are being prioritized as standard bugs as they're neither critical nor unreleased.

Reply:
sizable refactor, but i'm wondering if we should simplify queries here in favor of more server side filtering/sorting.  These filters got complicated very quickly now that we're addressing 3+ platforms and their unique business logic.  Might be more maintainable and testable to filter more in go (with hopefully minimal cpu/mem impact).

Reply:
Yep, always `AND`.

Reply:
> but i'm wondering if we should simplify queries here in favor of more server side filtering/sorting

Could you elaborate a bit on what you are thinking here, @mostlikelee?

Reply:
might be easier to show in code, i'll see if i can get something up in a draft PR

Reply:
> In any event, if that work isn't going to get done in this PR by @jahzielv then IMO this PR should go out as is

I agree (per my prior approval). But given our prior experiences with P1 issues stemming from disk encryption summaries/filters in our large MDM customer deployments, I do think it should be a priority to do at least some load testing or other time-boxed effort to ensure there's not a regression in 4.60 that we just haven't heard about yet.

Reply:
@gillespi314 so I'll merge this PR in. Sounds like we'll need a loadtest? It'll probably have to happen on Monday